### ISSUE TITLE
ENH: unsqueeze alias for expand_dims that mimics PyTorch's unsqueeze

### ISSUE THREAD

**Issue by adamconkey** [numpy/numpy#20511]
### Proposed new feature or change:

I would like to propose the addition of an "unsqueeze" function that works in the reverse direction of the existing [`squeeze`](https://numpy.org/doc/stable/reference/generated/numpy.squeeze.html) function. I believe the functionality would be identical to the [`expand_dims`](https://numpy.org/doc/stable/reference/generated/numpy.expand_dims.html) function, so perhaps I am just suggesting there be an alias `unsqueeze` for `expand_dims`.

The reason I propose this is the PyTorch API has largely mimicked the NumPy API when it comes to manipulating tensors, and PyTorch has an [`unsqueeze`](https://pytorch.org/docs/stable/generated/torch.unsqueeze.html) function. I find this more intuitive and compact than `expand_dims` and since I am frequently going back and forth between PyTorch and NumPy, I frequently have to remind myself what is the syntax for getting the `unsqueeze` functionality in NumPy. It's more compact because I can do

 ```
np_a = np.zeros(3)
torch_a = torch.tensor(np_a)
torch_a.unsqueeze(0)  # Gives tensor size (1, 3)
```
But I cannot do `np_a.expand_dims(0)` and instead have to do `np.expand_dims(np_a, 0)`. I think it would be more convenient to have the `expand_dims` functionality directly callable from an array (similar to PyTorch's `unsqueeze` for tensors) and then possibly have an alias of `unsqueeze` to be consistent with the PyTorch API.

Given [PyTorch's commitment to being familiar to NumPy users](https://github.com/pytorch/pytorch/issues/50344), I'm wondering if NumPy is also willing to adopt some PyTorch API that is possibly an improvement over the existing NumPy API. That `unsqueeze` is more intuitive to `expand_dims` may not be more widely accepted, but I thought I'd at least put it out there.

**Comment by WarrenWeckesser** [comment]
FYI: A few years ago @eric-wieser made a related suggestion that takes a different point of view: [Add np.collapse_dims as alias for np.squeeze? ](https://github.com/numpy/numpy/issues/8735)

**Comment by seberg** [comment]
Also, those involved in data-api's clearly thought that the NumPy naming is better or more prelevant: https://data-apis.org/array-api/latest/API_specification/manipulation_functions.html#expand-dims-x-axis

That means it is very likely that `pytorch` will adopt the NumPy naming, and IMO it will help no-one if all libraries grow both options.  (I would rather expect `pytorch` to eventually remove their `unsqueeze` name.)

**Comment by seberg** [comment]
Going to close this, since I don't think anyone from the core team took the bait, and it seems unlikely anyone will want to add an alias for this.
If anyone disagrees, please reopen.

**Comment by jimy-byerley** [comment]
It is sad that this quality-of-life issue has been closed due to
- there is already existing similar features despite less convenient
- other modules might adapt to numpy but not numpy to other modules

I think any of these reasons would be sufficient to prevent any project for any progress

I personally think it would be more sound to have the two reciprocal functions named in a similar way (`squeeze`/ `unsqueeze` or `collapse_dims`/`expand_dims`) and placed in the same place (both as `ndarray` methods or both as functions at numpy's root)
I prefer the names `squeeze`/`unsqueeze` since it is shorter and already enforced in pytorch. And I prefer having them as methods since it makes code more readable and because these are simply functions creating views and not actually generating or copying data.

**Comment by jimy-byerley** [comment]
For the projects I seen using numpy and unsqueezing operations (mostly deeplearning or computer vision code I must say, and sometimes linear algebra code), most users simply do not use `np.expand_dims` but rather prefer using the more complicated slicing syntax when it is possible (like for example `array[:,None]`). I guess it is because it is shorter than `np.expand_dims(array, 1)`
